GitUI/CommandsDialogs/BrowseDialog/GitStatusMonitor.cs

```diff
@@ -384,86 +324,78 @@ private void Update()
             // so we don't queue up a bunch of commands
             if (_commandIsRunning ||
 
-                // don't update status while repository is being modified by GitExt
-                // or while any git process is running, mostly repository status will change
-                // after these actions. Moreover, calling git status while other git command is performed
-                // can cause repository crash
+                // don't update status while repository is being modified by GitExt,
+                // repository status will change after these actions.
                 UICommandsSource.UICommands.RepoChangedNotifier.IsLocked ||
                 (GitVersion.Current.RaceConditionWhenGitStatusIsUpdatingIndex && Module.IsRunningGitProcess()))
             {
-                _statusIsUpToDate = false; // tell that computed status isn't up to date
+                _statusIsUpToDate = false; // schedule new update when command is finished
                 return;
             }
 
             _commandIsRunning = true;
             _statusIsUpToDate = true;
             _previousUpdateTime = Environment.TickCount;
 
+            // Schedule update every 5 min, even if we don't know that anything changed
+            CalculateNextUpdateTime(PeriodicUpdateInterval);
+
+            // capture a consistent state in the main thread
+            var statusCancellationToken = _statusCancellationToken;
```

Comment:
Actually we capture here only the current `_statusCancellationToken` not capturing the current `Module`. While it seems to be harmless in this particular situation, it gives a bad pattern to follow. The harmless corner case I see here is when the `_changedFilesProvider.Get() ` is invoked after the repo change. `_changedFilesProvider` will operate on the new repo, but the result of `Get` will be discarded because the token being cancelled. The result is valid and could be used. The loss is meaningless.

Reply:
Reverting ChangedFilesProvider changes would probably make this clearer

Reply:
Happy you are noticing that. 
I came across these talks related to this phenomenon
https://www.youtube.com/watch?v=JoTB2mcjU7w
https://www.youtube.com/watch?v=z9quxZsLcfo
Interestingly they all consider mocking as the last resort option.

Reply:
Whilst this certainly is a possibility, what is the likelihood of  this actually happening?

I haven't had a chance to watch the videos, but comments seem to support my drive to decouple.

Reply:
I reverted ChangedFilesProvider for now, readded the error handling and simplified some more. Commit chain is messy, but I will squash before merge. I expect that there will be further changes.

> Whilst this certainly is a possibility, what is the likelihood of this actually happening?

The problem pointed out was more to not set a pattern, low likelihood with no real drawback.

Removing ChangedFilesProvider makes this harder to test. I argue that GetStatusChangedFilesFromString() is tested anyway and ChangedFilesProvider adds a level of complexity without benefits.

Me and @RussKie have a different opinion on tests. RussKie is the 'tsar' and I try to adapt and be mostly quiet, it is much better that RussKie is happy than me slightly increases number of PRs. My (lazy) opinion is that a test should add something and RussKie that tests should always be added to not have a discussion.

Reply:
For me tests boost confidence and provide assurance when the code needs changing